File: scripts/check-env.js
#!/usr/bin/env node

/**
 * Environment Variables Checker
 * Validates that all required environment variables are present
 */

const fs = require('fs');
const path = require('path');

// Required environment variables
const REQUIRED_VARS = [
  'SUPABASE_URL',
  'SUPABASE_ANON_KEY', 
  'SUPABASE_SERVICE_ROLE_KEY',
  'API_BASE_URL',
  'JWT_SECRET'
];

// Optional but recommended variables
const RECOMMENDED_VARS = [
  'REDIS_URL',
  'NODE_ENV',
  'CORS_ORIGINS',
  'MAX_FILE_SIZE'
];

function loadEnvFile(filePath) {
  try {
    const envContent = fs.readFileSync(filePath, 'utf8');
    const envVars = {};
    
    envContent.split('\n').forEach(line => {
      line = line.trim();
      if (line && !line.startsWith('#') && line.includes('=')) {
        const [key, ...valueParts] = line.split('=');
        const value = valueParts.join('=').trim();
        if (key && value) {
          envVars[key] = value;
        }
      }
    });
    
    return envVars;
  } catch (error) {
    return null;
  }
}

function checkEnvironment() {
  console.log('🔍 Checking environment variables...\n');
  
  // Check if .env file exists
  const envPath = path.join(process.cwd(), '.env');
  const envExamplePath = path.join(process.cwd(), '.env.example');
  
  if (!fs.existsSync(envPath)) {
    console.log('❌ .env file not found');
    if (fs.existsSync(envExamplePath)) {
      console.log('💡 Run: cp .env.example .env');
      console.log('   Then edit .env with your actual values');
    }
    process.exit(1);
  }
  
  // Load environment variables
  const envVars = loadEnvFile(envPath);
  if (!envVars) {
    console.log('❌ Failed to read .env file');
    process.exit(1);
  }
  
  console.log('✅ .env file found and readable\n');
  
  // Check required variables
  let missingRequired = [];
  let emptyRequired = [];
  
  REQUIRED_VARS.forEach(varName => {
    if (!(varName in envVars)) {
      missingRequired.push(varName);
    } else if (!envVars[varName] || envVars[varName].includes('your-') || envVars[varName].includes('localhost')) {
      emptyRequired.push(varName);
    }
  });
  
  // Check recommended variables
  let missingRecommended = [];
  
  RECOMMENDED_VARS.forEach(varName => {
    if (!(varName in envVars)) {
      missingRecommended.push(varName);
    }
  });
  
  // Report results
  console.log('📋 Environment Variables Status:');
  console.log('================================');
  
  if (missingRequired.length === 0 && emptyRequired.length === 0) {
    console.log('✅ All required variables are present and configured');
  } else {
    if (missingRequired.length > 0) {
      console.log('❌ Missing required variables:');
      missingRequired.forEach(varName => {
        console.log(`   - ${varName}`);
      });
    }
    
    if (emptyRequired.length > 0) {
      console.log('⚠️  Required variables with placeholder values:');
      emptyRequired.forEach(varName => {
        console.log(`   - ${varName}: ${envVars[varName]}`);
      });
    }
  }
  
  if (missingRecommended.length > 0) {
    console.log('\n💡 Missing recommended variables:');
    missingRecommended.forEach(varName => {
      console.log(`   - ${varName}`);
    });
  }
  
  console.log('\n📊 Summary:');
  console.log(`   Required: ${REQUIRED_VARS.length - missingRequired.length - emptyRequired.length}/${REQUIRED_VARS.length} configured`);
  console.log(`   Recommended: ${RECOMMENDED_VARS.length - missingRecommended.length}/${RECOMMENDED_VARS.length} present`);
  
  // Exit with appropriate code
  if (missingRequired.length > 0 || emptyRequired.length > 0) {
    console.log('\n❌ Environment check failed');
    process.exit(1);
  } else {
    console.log('\n✅ Environment check passed');
    process.exit(0);
  }
}

// Run the check
checkEnvironment();

File: scripts/check_env.py
#!/usr/bin/env python3
"""
Environment Variables Checker (Python version)
Validates that all required environment variables are present
"""

import os
import sys
from pathlib import Path
from typing import Dict, List, Tuple


# Required environment variables
REQUIRED_VARS = [
    "SUPABASE_URL",
    "SUPABASE_ANON_KEY", 
    "SUPABASE_SERVICE_ROLE_KEY",
    "API_BASE_URL",
    "JWT_SECRET"
]

# Optional but recommended variables
RECOMMENDED_VARS = [
    "REDIS_URL",
    "NODE_ENV",
    "CORS_ORIGINS",
    "MAX_FILE_SIZE"
]


def load_env_file(file_path: Path) -> Dict[str, str]:
    """Load environment variables from a .env file."""
    env_vars = {}
    
    try:
        with open(file_path, 'r', encoding='utf-8') as f:
            for line in f:
                line = line.strip()
                if line and not line.startswith('#') and '=' in line:
                    key, *value_parts = line.split('=')
                    value = '='.join(value_parts).strip()
                    if key and value:
                        env_vars[key] = value
        return env_vars
    except FileNotFoundError:
        return {}
    except Exception as e:
        print(f"❌ Error reading .env file: {e}")
        return {}


def check_environment() -> bool:
    """Check environment variables and return True if all required vars are configured."""
    print("🔍 Checking environment variables...\n")
    
    # Check if .env file exists
    env_path = Path(".env")
    env_example_path = Path(".env.example")
    
    if not env_path.exists():
        print("❌ .env file not found")
        if env_example_path.exists():
            print("💡 Run: cp .env.example .env")
            print("   Then edit .env with your actual values")
        return False
    
    # Load environment variables
    env_vars = load_env_file(env_path)
    if not env_vars:
        print("❌ Failed to read .env file")
        return False
    
    print("✅ .env file found and readable\n")
    
    # Check required variables
    missing_required = []
    empty_required = []
    
    for var_name in REQUIRED_VARS:
        if var_name not in env_vars:
            missing_required.append(var_name)
        elif (not env_vars[var_name] or 
              'your-' in env_vars[var_name] or 
              'localhost' in env_vars[var_name]):
            empty_required.append(var_name)
    
    # Check recommended variables
    missing_recommended = []
    
    for var_name in RECOMMENDED_VARS:
        if var_name not in env_vars:
            missing_recommended.append(var_name)
    
    # Report results
    print("📋 Environment Variables Status:")
    print("================================")
    
    if not missing_required and not empty_required:
        print("✅ All required variables are present and configured")
    else:
        if missing_required:
            print("❌ Missing required variables:")
            for var_name in missing_required:
                print(f"   - {var_name}")
        
        if empty_required:
            print("⚠️  Required variables with placeholder values:")
            for var_name in empty_required:
                print(f"   - {var_name}: {env_vars[var_name]}")
    
    if missing_recommended:
        print("\n💡 Missing recommended variables:")
        for var_name in missing_recommended:
            print(f"   - {var_name}")
    
    print(f"\n📊 Summary:")
    configured_count = len(REQUIRED_VARS) - len(missing_required) - len(empty_required)
    present_count = len(RECOMMENDED_VARS) - len(missing_recommended)
    print(f"   Required: {configured_count}/{len(REQUIRED_VARS)} configured")
    print(f"   Recommended: {present_count}/{len(RECOMMENDED_VARS)} present")
    
    # Return success status
    success = len(missing_required) == 0 and len(empty_required) == 0
    
    if success:
        print("\n✅ Environment check passed")
    else:
        print("\n❌ Environment check failed")
    
    return success


def main():
    """Main entry point."""
    success = check_environment()
    sys.exit(0 if success else 1)


if __name__ == "__main__":
    main()
